### PR TITLE
Migrate to MSK

### PR DESCRIPTION
**Pull request by elenasuteva**
# Description

## What

As part of the kafka migration, this PR contains the changes for migrating from old kafka to MSK.

Using the Kafka-client-go/v3 library replaced the consumer and producer of Kafka messages, updated env vars.
Start producer and soncumer in the main function, update logger to v2 and update go version to 18.
Kafka consumer lag monitor healthcheck added.
Remove unused functions and tests are fixed.

## Why

https://financialtimes.atlassian.net/browse/UPPSF-3415

## Anything, in particular, you'd like to highlight to reviewers

Mention here sections of code which you would like reviewers to pay extra attention to .E.g

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [x] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,10 +33,9 @@ Options:
         --queue-addresses="http://%H:8080"                              Queue address ($Q_ADDR)
```
Comment (atanasdinov) [inline README.md:33]: Can we also update this to Kafka's port (9092)?

File: healthcheck.go [hunk 2]
```diff
@@ -45,34 +51,46 @@ func (h *HealthCheck) Health() func(w http.ResponseWriter, r *http.Request) {
...
-		Checker:          h.producer.ConnectivityCheck,
+		Checker:          h.checkIfKafkaIsReachableFromProducer,
+	}
+}
+
+func (h *HealthCheck) readQueueLagCheck() fthealth.Check {
+	return fthealth.Check{
+		ID:               "read-message-queue-lagging",
+		Name:             "Read Message Queue Is Not Lagging",
+		Severity:         2,
```
Comment (atanasdinov) [inline healthcheck.go:80]: I'd suggest to set the severity to `3` i.e. `low`.

File: healthcheck.go [hunk 3]
```diff
@@ -87,3 +105,27 @@ func gtgCheck(handler func() (string, error)) gtg.Status {
...
+	}
+	return "OK", nil
+}
+
+func (h *HealthCheck) checkIfConsumerIsLagging() (string, error) {
+	err := h.consumer.MonitorCheck()
+	if err != nil {
+		return "", err
+	}
+	return "OK", err
```
Comment (atanasdinov) [inline healthcheck.go:122]:
```suggestion
	return "OK", nil
```

File: helm/upp-next-video-content-collection-mapper/templates/deployment.yaml [hunk 4]
```diff
@@ -43,15 +43,17 @@ spec:
           value: "{{ .Values.env.APP_PORT }}"
         - name: Q_GROUP
           value: "{{ .Values.env.Q_GROUP }}"
-        - name: Q_READ_TOPIC
```
Comment (atanasdinov) [inline helm/upp-next-video-content-collection-mapper/templates/deployment.yaml:46]: How is this working? We can't stop reading from `NativeCmsPublicationEvents`?

File: helm/upp-next-video-content-collection-mapper/app-configs/upp-next-video-content-collection-mapper_eks_delivery.yaml [hunk 5]
```diff
@@ -7,5 +7,6 @@ env:
   APP_SYSTEM_CODE: upp-next-video-cc-mapper
   APP_PORT: 8080
   Q_GROUP: NextVideoContentCollectionMapper
-  Q_READ_TOPIC: NativeCmsPublicationEvents
```
Comment (atanasdinov) [inline helm/upp-next-video-content-collection-mapper/app-configs/upp-next-video-content-collection-mapper_eks_delivery.yaml:10]: We can't be removing this.

File: helm/upp-next-video-content-collection-mapper/app-configs/upp-next-video-content-collection-mapper_eks_delivery.yaml [hunk 6]
```diff
@@ -7,5 +7,6 @@ env:
   APP_SYSTEM_CODE: upp-next-video-cc-mapper
   APP_PORT: 8080
   Q_GROUP: NextVideoContentCollectionMapper
```
Comment (atanasdinov) [inline helm/upp-next-video-content-collection-mapper/app-configs/upp-next-video-content-collection-mapper_eks_delivery.yaml:9]: Could we set this to `upp-next-video-content-collection-mapper` for consistency purposes?

File: main.go [hunk 7]
```diff
@@ -75,68 +70,90 @@ func main() {
...
+
+		qh := queueHandler{
+			sc:              sc,
+			messageProducer: producer,
+			log:             log}
+
+		go func(consumer *kafka.Consumer) {
+			err := consumer.Close()
+			if err != nil {
+				log.WithError(err).Error("Producer could not stop")
```
Comment (atanasdinov) [inline main.go:145]:
```suggestion
				log.WithError(err).Error("Consumer could not stop")
```

**Review by atanasdinov — CHANGES_REQUESTED** [review]

File: healthcheck.go (hunk 2 above)
Comment (elenasuteva) [reply healthcheck.go:80]: And how do we determine the severity of the healthchecks in general?

File: helm/upp-next-video-content-collection-mapper/app-configs/upp-next-video-content-collection-mapper_eks_delivery.yaml (hunk 5 above)
Comment (elenasuteva) [reply helm/upp-next-video-content-collection-mapper/app-configs/upp-next-video-content-collection-mapper_eks_delivery.yaml:10]: Done by mistake, reverting

File: healthcheck.go (hunk 2 above)
Comment (atanasdinov) [reply healthcheck.go:80]: Depends on their importance. The lag monitoring is a status check which, when triggered, would either show us that there's a problem with the consumer connection or that there's an increase in the traffic to the cluster (which in itself is not an issue generally).

Comment (elenasuteva) [reply healthcheck.go:80]: Thank you!

File: main.go [hunk 8]
```diff
@@ -75,68 +70,90 @@ func main() {
...
+
+		producerConfig := kafka.ProducerConfig{
+			BrokersConnectionString: *kafkaAddress,
+			Topic:                   *writeTopic,
+			ConnectionRetryInterval: time.Minute,
+		}
 
-		hc := NewHealthCheck(qh.messageProducer, qh.messageConsumer, *appName, *appSystemCode, *panicGuide)
+		producer := kafka.NewProducer(producerConfig, log)
+		go func(producer *kafka.Producer) {
```
Comment (ilian2233) [inline main.go:130]: **issue:** The producer and consumer should be closed after `waitForSignal()` receives a signal. 
```suggestion
		defer go func(producer *kafka.Producer) {
```

**Review by ilian2233 — CHANGES_REQUESTED** [review]

File: main.go (hunk 8 above)
Comment (elenasuteva) [reply main.go:130]: Thank you, seems like I made a typo here.

File: README.md [hunk 9]
```diff
@@ -30,13 +30,12 @@ Options:
         --app-name="Next Video Content Collection Mapper"               Application name ($APP_NAME)
         --service-name="next-video-content-collection-mapper"           Service name ($SERVICE_NAME)
         --port="8080"                                                   Port to listen on ($APP_PORT)
-        --queue-addresses="http://%H:8080"                              Queue address ($Q_ADDR)
+        --queue-kafkaAddress="http://%H:9092"                              Queue address ($Q_ADDR)
```
Comment (atanasdinov) [inline README.md:33]:
```suggestion
        --queue-kafkaAddress="%H:9092"                              Queue address ($Q_ADDR)
```

Please also check whether the description is properly formatted (aligned to the right).

**Review by atanasdinov — COMMENTED** [review]
LGTM! Happy to approve once this minor documentation comment is addressed.

**Review by atanasdinov — APPROVED** [review]

**Comment by elenasuteva** [comment]
@atanasdinov should I decrease the replicaCount from 2 to 1?

**Review by ilian2233 — APPROVED** [review]

**Comment by atanasdinov** [comment]
> @atanasdinov should I decrease the replicaCount from 2 to 1?

Yes, it'd be better.

**Review by atanasdinov — APPROVED** [review]

**Pull request merged**